lib/timeout.rb

```diff
@@ -114,31 +149,32 @@ def self.create_timeout_thread
         requests.reject!(&:done?)
       end
     end
-    ThreadGroup::Default.add(watcher) unless watcher.group.enclosed?
+
+    if !watcher.group.enclosed? && (!defined?(Ractor.main?) || Ractor.main?)
+      ThreadGroup::Default.add(watcher)
+    end
+
     watcher.name = "Timeout stdlib thread"
     watcher.thread_variable_set(:"\0__detached_thread__", true)
     watcher
   end
   private_class_method :create_timeout_thread
 
   def self.ensure_timeout_thread_created
-    unless @timeout_thread and @timeout_thread.alive?
+    data = get_data
+
+    unless data.timeout_thread and data.timeout_thread.alive?
```

Comment:
```suggestion
    unless data.timeout_thread&.alive?
```
Ruby 2.6 seems to have `&.` so it seems good to use, and avoid an extra read